### PR TITLE
fix crash when DynamicFontAtSize freed by ref

### PR DESCRIPTION
**Pull request by sanikoyes**

### CODE REVIEW

**Comment by akien-mga** [comment]
Sounds like this would fix #15392.

File: scene/resources/dynamic_font.h [hunk 1]
```diff
@@ -68,7 +68,7 @@ class DynamicFontData : public Resource {
 	bool force_autohinter;
 
 	String font_path;
-	Map<CacheID, DynamicFontAtSize *> size_cache;
+	Map<CacheID, Ref<DynamicFontAtSize> > size_cache;
```
Comment (reduz) [inline scene/resources/dynamic_font.h]: This will create a cyclic reference and the font would never be freed

**Comment by poke1024** [comment]
Also note that `DynamicFontAtSize::~DynamicFontAtSize()` does a `font->size_cache.erase(id)`, so `size_cache.erase` will never contain any stale elements subject to two conditions:

* `DynamicFontAtSize ::id` is the same as when it was inserted and did not change (seems true)
* there's only a single thread at work here (hopefully true)

**Comment by akien-mga** [comment]
Superseded by 8daf5491.

**Pull request closed without merging**